R/tmb.R

```diff
@@ -277,12 +277,28 @@ h_mmrm_tmb_data <- function(formula_parts,
     subject_groups <- factor(rep(0L, n_subjects))
     n_groups <- 1L
   }
-
-  structure(
+  coordinates <- full_frame[formula_parts$visit_var]
+  if (formula_parts$cov_type %in% cov_type_spatial) {
+    lapply(
+      coordinates,
+      assert_numeric
+    )
+    coordinates_matrix <- as.matrix(coordinates)
+    visits_zero_inds <- 0L
+    n_visits <- 2L
+  } else {
+    assert(identical(ncol(coordinates), 1L))
+    assert_factor(coordinates[[1L]])
+    visits_zero_inds <- as.integer(coordinates[[1L]]) - 1L
+    coordinates_matrix <- as.matrix(visits_zero_inds, ncol = 1)
```

Comment:
do we really then use this in C++? if not, I would suggest to just pass an empty matrix

Reply:
currently, I don't know how to consolidate the coordinates and visits_zero_inds; but they should all represent the same meaning; I wonder one day we shall use coordinates to replace visits_zero_inds